src/groupByKey.ts

```diff
@@ -2,6 +2,24 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey(
+  items: object[],
+  key: string,
+): GroupsMap<typeof items> {
+  const obj: GroupsMap<typeof items> = {};
+  const keyArray = items.reduce((acc: string[], curr: object) => {
+    const valueKey: string = curr[key];
+
+    if (!acc.includes(valueKey)) {
+      acc.push(valueKey);
+    }
+
+    return acc;
+  }, []);
+
+  keyArray.forEach((e: string) => {
+    obj[e] = items.filter((curr: object) => curr[key] === e);
+  });
```

Comment:
You can do it just with one reduce just give object as initial value instead of array